### ISSUE TITLE
Managing Distribution Groups

### ISSUE THREAD

**Issue by bryancolligan** [ecederstrand/exchangelib#505]
looking to help manage Distribution Groups via the API.
https://docs.microsoft.com/en-us/exchange/client-developer/exchange-web-services/distribution-groups-and-ews-in-exchange#managing-distribution-groups-by-using-the-ews-managed-api-or-ews

Would like to help contribute and push to make this possible if it doesn't already exist.

Is this on the road map or would like to have this part of the project I can write the spec and code to make this happen.

**Comment by ecederstrand** [comment]
At a cursory glance, most of the services mentioned are supported, but may need additional work. The only one I see missing is `ExpandDL`.

I have no plans to implement this, but you're most welcome to identify the missing features and contribute changes. Feel free to discuss and ask questions here or in new, more specific issues.

As a start, let's list the functionality needed and a suggested API for working with distribution groups as Python example code.

**Comment by ba1dr** [comment]
Below is what I have done for my needs so far.

To inject the new code to existing library I apply:
```python
from features import exchangelib_patch
exchangelib_patch()
```
Anyone may take this code and make a pull request.

File `features.py`:
```python

from exchangelib.protocol import Protocol
from exchangelib.services import EWSService, MNS
from exchangelib.properties import EWSElement, Mailbox
from exchangelib.util import create_element, set_xml_value
from exchangelib.fields import EmailAddressField


class DLMailbox(EWSElement):
    # Like Mailbox, but with slightly different attributes
    ELEMENT_NAME = 'Mailbox'
    NAMESPACE = MNS
    FIELDS = [
        EmailAddressField('email_address', field_uri='EmailAddress', is_required=True),
    ]

    __slots__ = ('email_address', )

    def __hash__(self):
        return hash(self.email_address.lower())

    @classmethod
    def from_mailbox(cls, mailbox):
        if not isinstance(mailbox, Mailbox):
            raise ValueError("'mailbox' %r must be a Mailbox instance" % mailbox)
        return cls(email_address=mailbox.email_address)


class ExpandDL(EWSService):
    """
    https://docs.microsoft.com/en-us/exchange/client-developer/web-service-reference/expanddl-operation
    """
    SERVICE_NAME = 'ExpandDL'
    element_container_name = '{%s}DLExpansion' % MNS

    def call(self, mailbox):
        elements = self._get_elements(payload=self.get_payload(mailbox=mailbox))
        return [Mailbox.from_xml(elem=elem, account=None) for elem in elements]

    def get_payload(self, mailbox):
        expanddl = create_element('m:%s' % self.SERVICE_NAME)
        set_xml_value(expanddl, mailbox, version=self.protocol.version)
        return expanddl


def expand_dl(self, email_address):
    return ExpandDL(protocol=self).call(mailbox=DLMailbox(email_address=email_address))


def exchangelib_patch():
    Protocol.expand_dl = expand_dl
```

To call I use `config.protocol.expand_dl("group1@domain.com")`. It returns all entries as `Mailbox` objects array, but **may contain other groups**, deep expanding is not implemented.

**Comment by ecederstrand** [comment]
Fixes by #560
